### ISSUE TITLE
Expose `isEmail` or something similar to `renderField`

### ISSUE THREAD

**Issue by danielweinmann** [SeasonedSoftware/remix-forms#83]
Right now, it's not that simple to infer if a field is a `ZodString` with an `email` check from inside `renderField`.

We could make `shapeInfo` parse this information and expose it somehow, to help users who want to render email fields differently from other inputs do so. In the meantime, I'm sharing a partial solution on #81.

**Comment by danielweinmann** [comment]
Actually, after reading #76, I think we should expose a prop with the "HTML type" of the field. That way, we can allow people to render HTML types using `renderField` if they want to.

Personally, I don't like mixing react-hook-form's client-side validations with standard browser validations. That's why I don't want to make this the default behavior. But if we infer the "HTML type" on `shapeInfo` and expose it to `renderField` I think we'll empower people to decide for themselves without too much extra work.

**Comment by azzzy** [comment]
> Personally, I don't like mixing react-hook-form's client-side validations with standard browser validations.

@danielweinmann, providing the correct input type to the DOM element has accessibility benefits and in my opinion is a better default.
The built-in browser validations can be disabled by adding the `novalidate` html attribute, wouldn't this help?

**Comment by danielweinmann** [comment]
@azzzy, agreed! Using the correct input types + `novalidate` seems like the best default for us. Thanks for the suggestion!
